packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx

```diff
@@ -0,0 +1,282 @@
+                  + taxes /month
+                </Typography>
+              </Box>
+            </ProductContainer>
+            <Box>
+              <Typography variant="caption" color="text.primary">
+                By completing your purchase, you consent to BaseApp storing your payment method for
+                future charges. You can change your payment method at any time in your account
+                settings.
+              </Typography>
+            </Box>
+            <Box>
+              <StyledLoadingButton
+                variant="contained"
+                color="primary"
+                loading={isCreatingSubscription || isConfirmCardPaymentProcessing}
+                onClick={isRetry ? handleRetry : handlePlaceOrder}
+                disabled={
+                  isCreatingSubscription ||
+                  confirmationModalOpen ||
+                  isConfirmCardPaymentProcessing ||
+                  paymentMethods?.length === 0
+                }
+              >
+                {isRetry ? 'Retry' : 'Place Order'}
+              </StyledLoadingButton>
+            </Box>
+          </Box>
+        </Grid>
+
+        <Grid item xs={12} sm={6}>
+          <Box sx={{ maxWidth: 400 }}>
+            <Box
+              sx={{
+                display: 'flex',
+                flexDirection: 'column',
+                gap: 2,
+              }}
+            >
+              <Typography variant="subtitle2">Payment</Typography>
+              <Divider variant="fullWidth" sx={{ backgroundColor: 'divider' }} />
+              {elements && stripe && (
+                <PaymentDropdown
+                  customerId={checkoutCustomerId}
+                  paymentMethods={paymentMethods}
+                  selectedPaymentMethodId={selectedPaymentMethodId}
+                  setSelectedPaymentMethodId={setSelectedPaymentMethodId}
+                  elements={elements}
+                  stripe={stripe}
+                  isAddCardModalOpen={isAddCardModalOpen}
+                  setIsAddCardModalOpen={setIsAddCardModalOpen}
+                  handleSetupSuccess={handleSetupSuccess}
+                />
+              )}
+              {shouldRenderAddressElement ? (
+                <>
+                  <Typography variant="subtitle2">Billing Address</Typography>
+
+                  <Divider
+                    variant="fullWidth"
+                    sx={{ backgroundColor: 'divider', color: 'divider' }}
+                  />
+                  <Box
+                    // make sure the AddressElement update the address when changing the payment method
+                    key={`address-${selectedPaymentMethodId}-modal-${isAddCardModalOpen ? 'open' : 'closed'}`}
+                  >
+                    <AddressElement options={addressOptions} />
+                  </Box>
+                </>
+              ) : null}
+            </Box>
+          </Box>
+        </Grid>
+      </Grid>
+      <ConfirmationSubscriptionModal
```

Comment:
Can we make this ConfirmationSubscriptionModal more "customizable"?
Can you add on the props something like ConfirmationSubscriptionModal = DefaultConfirmationSubscriptionModal so we can allow people to build their own modal if they want?